### PR TITLE
Persistent DHT cache

### PR DESCRIPTION
**Pull request by AndreiRegiani**
```bash

cat /Users/user/holepunchto/pear/pear/dht.json

{
    "nodes":[
        { "host": "127.0.0.1", "port": 49001 },
        { "host": "127.0.0.2", "port": 49002 }
    ]
}

```

### CODE REVIEW

**Comment by mafintosh** [comment]
Store should be named dht instead of known-nodes and the key you set simply nodes instead of dht

File: subsystems/sidecar/index.js [hunk 1]
```diff
@@ -892,6 +894,20 @@ class Sidecar extends ReadyResource {
...
   }
 
+  #setKnownNodes (dht) {
+    if (dht) knownNodes.set('dht', dht)
+  }
+
+  #getKnownNodes () {
+    let dht = this.swarm.dht.toArray({ limit: KNOWN_NODES_LIMIT })
+    if (dht) {
+      this.#setKnownNodes(dht)
```
Comment (mafintosh) [inline subsystems/sidecar/index.js:904]: Dont think it should set them here, just on teardown explicitly

File: subsystems/sidecar/index.js [hunk 2]
```diff
@@ -892,6 +894,20 @@ class Sidecar extends ReadyResource {
     yield * preferences.entries()
   }
 
+  #setKnownNodes (dht) {
+    if (dht) knownNodes.set('dht', dht)
```
Comment (davidmarkclements) [inline subsystems/sidecar/index.js:898]: avoid redirection without purpose - just set inline where you need it

File: subsystems/sidecar/index.js (hunk 1 above)
Comment (davidmarkclements) [reply subsystems/sidecar/index.js:904]: so function isn't needed then, just do this all inline in shutdown

**Review by mafintosh — APPROVED** [review]

**Pull request closed without merging**